package NYT.NCypressServer.NProto;

import "yt_proto/yt/core/misc/proto/guid.proto";

import "yt_proto/yt/core/ytree/proto/attributes.proto";

import "yt/server/master/cypress_server/proto/portal_manager.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqCreateScion
{
    required NYT.NProto.TGuid scion_node_id = 1;
    required NYT.NProto.TGuid rootstock_node_id = 2;
    optional NYT.NProto.TGuid parent_node_id = 14;

    required NYT.NProto.TGuid account_id = 3;

    required NYT.NYTree.NProto.TAttributeDictionary explicit_node_attributes = 4;
    required NYT.NYTree.NProto.TAttributeDictionary inherited_node_attributes = 5;

    required string path = 6;

    optional string key = 8;

    required bytes effective_acl = 9;
    optional bytes direct_acl = 10;
    optional bool inherit_acl = 11;

    optional string effective_annotation = 12;
    optional string effective_annotation_path = 13;
}

////////////////////////////////////////////////////////////////////////////////

message TReqRemoveRootstock
{
    required NYT.NProto.TGuid rootstock_node_id = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqSynchronizeScions
{
    repeated TPortalExitInfo scion_infos = 1;
}

////////////////////////////////////////////////////////////////////////////////
